cmd/lakefs/cmd/migrate.go

```diff
@@ -94,34 +94,41 @@ func migrateToACL(ctx context.Context, kvStore kv.Store, cfg *config.Config, log
 		_, _ = fmt.Fprintf(os.Stderr, "Failed to upgrade RBAC policies to ACLs: %s\n", err)
 		os.Exit(1)
 	}
-
+	hasWarnings := false
 	for _, w := range groupReports {
-		fmt.Printf("GROUP %s\n\tACL: %s\n", w.GroupID, ReportACL(w.ACL))
+		if printMessages {
+			fmt.Printf("GROUP %s\n\tACL: %s\n", w.GroupID, ReportACL(w.ACL))
+		}
 		if w.Warn != nil {
-			var multi *multierror.Error
-			if errors.As(w.Warn, &multi) {
-				multi.ErrorFormat = func(es []error) string {
-					points := make([]string, len(es))
-					for i, err := range es {
-						points[i] = fmt.Sprintf("* %s", err)
+			hasWarnings = true
+			if printMessages {
+				var multi *multierror.Error
+				if errors.As(w.Warn, &multi) {
+					multi.ErrorFormat = func(es []error) string {
+						points := make([]string, len(es))
+						for i, err := range es {
+							points[i] = fmt.Sprintf("* %s", err)
+						}
+						plural := "s"
+						if len(es) == 1 {
+							plural = ""
+						}
+						return fmt.Sprintf(
+							"%d change%s:\n\t%s\n",
+							len(points), plural, strings.Join(points, "\n\t"))
 					}
-					plural := "s"
-					if len(es) == 1 {
-						plural = ""
-					}
-					return fmt.Sprintf(
-						"%d change%s:\n\t%s\n",
-						len(points), plural, strings.Join(points, "\n\t"))
 				}
+				fmt.Println(w.Warn)
 			}
-			fmt.Println(w.Warn)
 		}
 		fmt.Println()
 	}
 	for _, username := range usersWithPolicies {
-		fmt.Printf("USER (%s)  detaching directly-attached policies\n", username)
+		if printMessages {
+			fmt.Printf("USER (%s)  detaching directly-attached policies\n", username)
+		}
```

Comment:
I still don't understand the requirement: why are these treated specially?  Can't we just put these on warnings?

Reply:
Currently, warnings have an ACL-specific logic on printing. Tried using warnings at first which resulted, in making them fit both group warnings and user-policy warnings felt like over-complicating things, but looks like I ended up doing that anyway.